ui/pages/settings/networks-tab/networks-form/networks-form.js

```diff
@@ -129,9 +131,11 @@ const NetworksForm = ({
     // was possible in versions <8.1 of the extension.
     // Basically, we always want to be able to overwrite an invalid chain ID.
     const chainIdIsUnchanged =
-      typeof selectedNetwork.chainId === 'string' &&
-      selectedNetwork.chainId.toLowerCase().startsWith('0x') &&
-      chainId === getDisplayChainId(selectedNetwork.chainId);
+      typeof selectedNetwork.caipChainId === 'string' &&
+      isEthCaipChainId(selectedNetwork.caipChainId) &&
+      chainId && // getEthChainIdHexFromCaipChainId should be updated to handle empty input
```

Comment:
reminder to check on this once getEthChainIdHexFromCaipChainId gets more nailed down